src/main/java/uk/gov/ida/eventemitter/AmazonSqsClient.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.ida.eventemitter;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.model.SendMessageRequest;
+import com.google.inject.Inject;
+
+import javax.inject.Named;
+
+public class AmazonSqsClient implements SqsClient {
+
+    private final AmazonSQS sqs;
+    private final String queueUrl;
+
+    @Inject
+    public AmazonSqsClient(final AmazonSQS sqs,
+                           @Named("QueueName") final String queueName) {
+        this.sqs = sqs;
+        this.queueUrl = sqs.getQueueUrl(queueName).getQueueUrl();
```

Comment:
It is a little surprising to find external HTTP requests in a constructor (can unexpectedly make the constructor slow / blocking / fail)

Could we just pass in the `queueUrl`directly?

Reply:
This has been done.